packages/invoices/src/OutboundShipment/ListView/ListView.tsx

```diff
@@ -35,9 +36,20 @@ export const OutboundShipmentListViewComponent: FC = () => {
 
   const { data, isError, isLoading, sort, pagination, filter } =
     useOutbound.document.list();
-  const { onChangeSortBy, sortBy } = sort;
+  const {
+    urlQuery,
+    updateSortQuery,
+    updatePaginationQuery,
+    updateFilterQuery,
+    onChangeUrlQuery,
+  } = useHandleQueryParams({ sort, pagination, filterKey: 'otherPartyName' });
+  const { sortBy } = sort;
   useDisableOutboundRows(data?.nodes);
 
+  useEffect(() => {
+    onChangeUrlQuery(columns, filter);
+  }, [urlQuery]);
+
```

Comment:
I don't like having this `useEffect` in here. Would prefer it was all handled inside `useHandleQueryParams`. Unfortunately`onChangeUrlQuery` needs access to the *current* columns state, so has to be passed in when run, can't use the value of columns when the hook is first instantiated.

Any suggestions?